python/plugins/processing/gui/DestinationSelectionPanel.py

```diff
@@ -288,9 +292,4 @@ def getValue(self):
         if isinstance(self.parameter, QgsProcessingParameterFolderDestination):
             return self.leText.text()
 
-        if isinstance(self.parameter, QgsProcessingParameterFileDestination):
```

Comment:
Won't removing this break file encoding support?

Reply:
I moved this in AlgorithmDialog, because in case of model output default value, I prefer having all times a string value, no need for a QgsProcessingOutputLayerDefinition in case we do not run the algortihm.

Reply:
I disagree - I think that if encoding isn't exposed in the default setting then it's really going to harm our international users who require correct encoding handling. I believe it's really just a matter of reverting these changes and then adding support for QgsProcessingOutputLayerDefinition in QgsXmlUtils::writeVariant and QgsXmlUtils::readVariant

Reply:
Adding createOptions to QgsProcessingModelOutput would make more sense to me on core API side.